### PR TITLE
Add reader macros for SBCL

### PR DESCRIPTION
**Pull request by equwal**
Everything is escaped with #+ and #- for sbcl. Read the readme for more infomation about them.

### CODE REVIEW

**Comment by equwal** [comment]
So the tests fail but it works better now. Even someone on SBCL not using the reader macros has access to the functions defined internally with them.  A little more work to make the reader macros compatible with strings would make the entire library function.

**Comment by equwal** [comment]
Well now the whole library functions, but the reader macros will still break when the user does strings or other reader macros internally. I didn't really want to reimplement the whole reader, but this is good enough to let me use the code from the book without having to convert everything to `with-gensyms` and `once-only` by hand which is tedious.

File: automatic-gensyms.lisp [hunk 1]
```diff
@@ -0,0 +1,187 @@
+;;;; -*- Mode: LISP; Syntax: COMMON-LISP; Package: LET-OVER-LAMBDA; Base: 10 -*- file: let-over-lambda.lisp
+
+;;;; Copyleft Spenser Truex "Equwal" 2018
```
Comment (thephoeron) [inline automatic-gensyms.lisp]: Could you remove the "copyleft" declaration, please? It's an unfair limitation of license that violates the intent of the original release, of which this is a derivative work.

Comment (equwal) [reply automatic-gensyms.lisp]: My mistake.

**Comment by equwal** [comment]
I'm going to have to cancel my attempt at this thing.

**Pull request closed without merging**